### ISSUE TITLE
Panic caused by Probe.getUserName

### ISSUE THREAD

**Issue by tghartland** [neuvector/neuvector#594]
**Environment**
Platform: Kubernetes (kubeadm managed)
Kubernetes/Platform Version(s): 1.24.3

**Describe the bug**
Concurrent writes to a map of user IDs can cause a panic, forcing neuvector enforcer to restart.

**To Reproduce**
Can't force it to happen as it's a concurrency issue, have only observed.
Not sure if it correlates with something happening inside one of the other pods on the node (e.g a process fork).
Seen in version 5.0.2, but looks like the relevant code is unchanged in the master branch.
Logs:
```
fatal error: concurrent map writes
2023-01-04T13:51:04.516|ERRO|AGT|probe.(*Probe).checkUserGroup_uidChange: Get User name fail - euid=1217612 ruid=1217612

goroutine 70463 [running]:
runtime.throw(0x1b6d06f, 0x15)
        /usr/local/go/src/runtime/panic.go:1116 +0x72 fp=0xc00095afb8 sp=0xc00095af88 pc=0x9cf752
runtime.mapassign_fast64(0x19b6420, 0xc0009881e0, 0x0, 0x0)
        /usr/local/go/src/runtime/map_fast64.go:101 +0x323 fp=0xc00095aff8 sp=0xc00095afb8 pc=0x9ac9c3
github.com/neuvector/neuvector/share/osutil.GetAllUsers(0x286a5, 0xc0009881e0, 0x12944c, 0x2c081a0, 0x0, 0x0)
        /go/src/github.com/neuvector/neuvector/share/osutil/user_linux.go:111 +0x2fc fp=0xc00095b130 sp=0xc00095aff8 pc=0x14f9afc
github.com/neuvector/neuvector/agent/probe.(*Probe).getUserName(0xc0001c6540, 0x286a5, 0x12944c, 0x203000, 0x0)
        /go/src/github.com/neuvector/neuvector/agent/probe/process.go:1286 +0xe0 fp=0xc00095b198 sp=0xc00095b130 pc=0x15416b0
github.com/neuvector/neuvector/agent/probe.(*Probe).checkUserGroup_uidChange(0xc0001c6540, 0xc003b02240, 0xc0004ca0e0, 0x1, 0x1, 0x400, 0x7f1f24938600, 0x9bb745)
        /go/src/github.com/neuvector/neuvector/agent/probe/process.go:817 +0x60 fp=0xc00095b4b8 sp=0xc00095b198 pc=0x153d590
github.com/neuvector/neuvector/agent/probe.(*Probe).rootEscalationCheck_uidChange(0xc0001c6540, 0xc003724900, 0xc0004ca0e0)
        /go/src/github.com/neuvector/neuvector/agent/probe/process.go:917 +0x3aa fp=0xc00095bfc8 sp=0xc00095b4b8 pc=0x153df4a
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00095bfd0 sp=0xc00095bfc8 pc=0xa00881
created by github.com/neuvector/neuvector/agent/probe.(*Probe).handleProcUIDChange
        /go/src/github.com/neuvector/neuvector/agent/probe/process.go:1277 +0x2a6
```

The map write:
https://github.com/neuvector/neuvector/blob/a552e5e0ea0a03c7f7b2c7de015f862db7835804/share/osutil/user_linux.go#L111
Where the map comes from (`c.userns.users`):
https://github.com/neuvector/neuvector/blob/a552e5e0ea0a03c7f7b2c7de015f862db7835804/agent/probe/process.go#L1286
And where the goroutine is created:
https://github.com/neuvector/neuvector/blob/a552e5e0ea0a03c7f7b2c7de015f862db7835804/agent/probe/process.go#L1277

After the full panic trace, there is a graceful shutdown/restart of the agent.
```
2023-01-04T13:51:04|MON|Process agent exit status 2, pid=2867591
2023-01-04T13:51:04|MON|Kill dp with signal 15, pid=2867802
2023-01-04T13:51:04.713Z [INFO]  agent.client: client starting leave
2023-01-04T13:51:04|DEBU|dp0|dp_data_thr: dp thread exits
...
2023-01-04T13:51:09|MON|Start agent, pid=166505
2023-01-04T13:51:09.749|INFO|AGT|main.main: START - version=v5.0.2
```

The same panic then happened three more times in the next ~10 minutes.
Full logs for that time are attached as a file:
[neuvector-enforcer-pod-8g984.log](https://github.com/neuvector/neuvector/files/10345366/neuvector-enforcer-pod-8g984.log)

**Expected behavior**
The map should be guarded against concurrent writes.

**Screenshots**
.

**UI Issue**
.

**Additional context**
.

**Comment by jaimeyu-suse** [comment]
I can take a closer look. I hit something similar recently but I don't think it applies here.

At first glance, it looks like we're missing a lock here around here https://github.com/neuvector/neuvector/blob/a552e5e0ea0a03c7f7b2c7de015f862db7835804/agent/probe/process.go#L917 for this particular crash.

**Comment by jaimeyu-suse** [comment]
Hi,
I created a PR that adds locks around the area that has the problem: https://github.com/neuvector/neuvector/pull/597

Thank you for helping us make the product better. It will go through its QA process and will be in the next release.

**Comment by becitsthere** [comment]
This will be fixed in 5.1.1 release.
